src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -1311,9 +1311,14 @@ GenTree* Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
                 GenTree*    trueMask        = comp->gtNewSimdAllTrueMaskNode(simdBaseJitType, simdSize);
                 GenTree*    trueVal         = node;
                 GenTree*    falseVal        = comp->gtNewZeroConNode(simdType);
+                var_types   nodeType        = simdType;
+                if (HWIntrinsicInfo::ReturnsPerElementMask(node->GetHWIntrinsicId()))
```

Comment:
I need to test this out to make sure this is correct. I will pull your branch once  https://github.com/dotnet/runtime/pull/103288 is merged.

Reply:
I am seeing the test `DOTNET_TieredCompilation=0`:

```
Assert failure(PID 18916 [0x000049e4], Thread: 28152 [0x6df8]): Assertion failed 'unreached' in 'JIT.HardwareIntrinsics.Arm._Sve.SimpleBinaryOpTest__Sve_AbsoluteCompareGreaterThan_float:ConditionalSelect_MethodMask():this' during 'Generate code' (IL size 109; hash 0x5f5a6ecd; FullOpts)

    File: D:\git\runtime\src\coreclr\jit\emitarm64sve.cpp:4396
    Image: D:\kpathak\Core_Root_absolutecompare\Core_Root\corerun.exe
```

This is missing a piece where if the `cndSelNode` is of type `TYP_MASK`, then the `falseValue` should also be `TYP_MASK`. Today, we do not support ZeroConNode of `TYP_MASK`, so I was thinking of wrapping `falseVal` with `ConvertVectorToMask()`, but for some reason that was failing in LIR validation that I didn't get a chance to verify. I will look tomorrow.

Reply:
@a74nh, @mikabl-arm  - I have fixed code to handle `falseValue` of `TYP_MASK`. PTAL.

Stress tests are passing: https://gist.github.com/kunalspathak/95ddd4913a6642bcbda7ddff9c33f752

The failure case `RunLoadMask` is because we are not preserving `p0` across call (known problem).

Reply:
LGTM